### ISSUE TITLE
Not working for CSharp code, for example

### ISSUE THREAD

**Issue by ghost** [kcmr/code-sample#6]
I tried to use a template like this:

    <code-sample type="csharp">
        <template preserve-content>
            void f()
            {
                int sum = 0;
                for (int i = 0; i < 100; i++)
                {
                    List<int> result = CallFunction(i);
                }
            }
        </template>
    </code-sample>

but I (strangely) obtain a result the '<' symbol in the 'for' is replaced by the &lt; and it creates a closing tag '</int>', corresponding erroneously to the List<int> definition, as if it was a tag.
Am I missing anything?
Thanks
Andrea

**Comment by kcmr** [comment]
Hi @andreabioli 
Thank you for submitting the issue.

There are two problems here:
- csharp language is not included in the highlight pack bundled with this component. The type to use had to be "cs" instead of "csharp".
- as the component is meant mainly for HTML (Web Components), JavaScript and CSS, there is a function that replaces HTML opening and closing tags.

I'm going to think in a solution (maybe a boolean attribute) to prevent replacing HTML opening and closing tags.

**Comment by kcmr** [comment]
After some investigation, the magical closing tag has nothing to do with the replacements made. The implicit closing tag is inserted automatically by HTML, so the only solution is to use the entitized chars inside `<code-sample>`.

There is a [similar issue on highlight.js](https://github.com/isagalaev/highlight.js/issues/1280)

I've just released a new version (v0.4.2) that replaces entitized characters (`&lt;` and `&gt;`) by `<` and `>` respectively, so now you can use that chars to get the right code, although the C# syntax is not included yet (I'll include more optional languages in the future).

The following code should produce the right output:

```html
<code-sample type="cs">
  <template>
    void f()
      {
        int sum = 0;
        for (int i = 0; i < 100; i++)
          {
            List&lt;int&gt; result = CallFunction(i);
          }
      }
  </template>
</code-sample>
```

You don't need to use the `preserve-content` attribute unless you're using the component inside another component.

I'm closing the issue for now, as there is no easy way to prevent that closing tags without using entity chars.
